File: Accout/Models/Account.swift
import SwiftUI
import SwiftData

/// 账户类型：资产（现金/存款/投资/借出）与负债（房贷/车贷/信用卡/借入等）
enum AccountKind: String, CaseIterable, Identifiable, Codable {
    // 资产
    case cash = "现金"
    case deposit = "存款"
    case investment = "投资"
    case receivable = "借出"
    // 负债
    case mortgage = "房贷"
    case carLoan = "车贷"
    case creditCard = "信用卡"
    case borrowing = "借入"
    case otherDebt = "其他负债"

    var id: String { rawValue }

    var isLiability: Bool {
        switch self {
        case .mortgage, .carLoan, .creditCard, .borrowing, .otherDebt: true
        default: false
        }
    }

    static var assetKinds: [AccountKind] { allCases.filter { !$0.isLiability } }
    static var liabilityKinds: [AccountKind] { allCases.filter(\.isLiability) }

    var icon: String {
        switch self {
        case .cash: "yensign.circle.fill"
        case .deposit: "building.columns.fill"
        case .investment: "chart.line.uptrend.xyaxis"
        case .receivable: "arrow.up.right.circle.fill"
        case .mortgage: "house.fill"
        case .carLoan: "car.fill"
        case .creditCard: "creditcard.fill"
        case .borrowing: "arrow.down.left.circle.fill"
        case .otherDebt: "doc.text.fill"
        }
    }

    var color: Color {
        switch self {
        case .cash: .green
        case .deposit: .blue
        case .investment: .purple
        case .receivable: .mint
        case .mortgage: .brown
        case .carLoan: .cyan
        case .creditCard: .red
        case .borrowing: .orange
        case .otherDebt: .gray
        }
    }
}

/// 资产/负债账户：balance 是资产现值或负债欠款；投资类可记 costBasis（投入本金）算收益
@Model
final class Account {
    var name: String
    var kindRaw: String
    var balance: Decimal
    var costBasis: Decimal
    var note: String
    var createdAt: Date
    var updatedAt: Date

    init(name: String, kind: AccountKind, balance: Decimal, costBasis: Decimal = 0, note: String = "") {
        self.name = name
        self.kindRaw = kind.rawValue
        self.balance = balance
        self.costBasis = costBasis
        self.note = note
        self.createdAt = .now
        self.updatedAt = .now
    }

    var kind: AccountKind {
        get { AccountKind(rawValue: kindRaw) ?? .cash }
        set { kindRaw = newValue.rawValue }
    }

    /// 投资收益（仅投资类且录入过本金时有意义）
    var profit: Decimal? {
        guard kind == .investment, costBasis > 0 else { return nil }
        return balance - costBasis
    }
}

File: models/account.ts
/**
 * 账户类型 - 对应 iOS AccountKind
 */
export enum AccountKind {
  // 资产
  Cash = '现金',
  Deposit = '存款',
  Investment = '投资',
  Receivable = '借出',
  // 负债
  Mortgage = '房贷',
  CarLoan = '车贷',
  CreditCard = '信用卡',
  Borrowing = '借入',
  OtherDebt = '其他负债',
}

/** 是否为负债类型 */
export function isLiability(kind: AccountKind): boolean {
  return [
    AccountKind.Mortgage,
    AccountKind.CarLoan,
    AccountKind.CreditCard,
    AccountKind.Borrowing,
    AccountKind.OtherDebt,
  ].includes(kind)
}

/** 资产类型列表 */
export function assetKinds(): AccountKind[] {
  return [AccountKind.Cash, AccountKind.Deposit, AccountKind.Investment, AccountKind.Receivable]
}

/** 负债类型列表 */
export function liabilityKinds(): AccountKind[] {
  return [AccountKind.Mortgage, AccountKind.CarLoan, AccountKind.CreditCard, AccountKind.Borrowing, AccountKind.OtherDebt]
}

/** 账户类型图标 */
export function accountKindIcon(kind: AccountKind): string {
  const map: Record<string, string> = {
    [AccountKind.Cash]: '💴',
    [AccountKind.Deposit]: '🏦',
    [AccountKind.Investment]: '📊',
    [AccountKind.Receivable]: '↗️',
    [AccountKind.Mortgage]: '🏡',
    [AccountKind.CarLoan]: '🚙',
    [AccountKind.CreditCard]: '💳',
    [AccountKind.Borrowing]: '↙️',
    [AccountKind.OtherDebt]: '📄',
  }
  return map[kind] || '📄'
}

/** 账户类型颜色 */
export function accountKindColor(kind: AccountKind): string {
  const map: Record<string, string> = {
    [AccountKind.Cash]: '#22C55E',
    [AccountKind.Deposit]: '#3B82F6',
    [AccountKind.Investment]: '#8B5CF6',
    [AccountKind.Receivable]: '#06B6D4',
    [AccountKind.Mortgage]: '#78716C',
    [AccountKind.CarLoan]: '#06B6D4',
    [AccountKind.CreditCard]: '#EF4444',
    [AccountKind.Borrowing]: '#F97316',
    [AccountKind.OtherDebt]: '#9CA3AF',
  }
  return map[kind] || '#9CA3AF'
}

/**
 * 资产负债账户 - 对应 iOS Account SwiftData Model
 */
export interface Account {
  id: string
  name: string
  kind: AccountKind
  balance: number
  costBasis: number        // 投资本金
  note: string
  createdAt: number
  updatedAt: number
}

/** 创建新账户 */
export function createAccount(partial?: Partial<Account>): Account {
  const now = Date.now()
  return {
    id: generateId(),
    name: '',
    kind: AccountKind.Deposit,
    balance: 0,
    costBasis: 0,
    note: '',
    createdAt: now,
    updatedAt: now,
    ...partial,
  }
}

/** 计算投资收益（仅投资类有意义） */
export function accountProfit(account: Account): number | null {
  if (account.kind !== AccountKind.Investment || account.costBasis <= 0) return null
  return account.balance - account.costBasis
}

function generateId(): string {
  return 'ac_' + Date.now().toString(36) + Math.random().toString(36).slice(2, 8)
}
